sale_global_discount/hooks.py

```diff
@@ -0,0 +1,35 @@
+from odoo.tools.sql import column_exists
+
+
+def _pre_init_global_discount_fields(cr):
+    if not column_exists(cr, "sale_order", "amount_global_discount"):
+        cr.execute(
+            """
+                ALTER TABLE "sale_order"
+                ADD COLUMN "amount_global_discount" double precision DEFAULT 0
```

Comment:
Remove this default after the creation, or it will apply always when no value is set, having potentially side effects:

https://github.com/OCA/openupgradelib/blob/f45dcfe2a3e9661a4104298e9358dedb8aa65b2f/openupgradelib/openupgrade.py#L2811-L2822